Responsive Site

OzzieOrca opened the pull request:
So I was thinking about the questions page and how it was not very mobile friendly both because the drag and drop didn't work and because nothing scaled down to the phone size. I was going to prototype it a little bit and see what ideas I had but it turned into much more than that with a new questions toolbar, pages and their blocks in panels and toolbars for each of those blocks.

![image](https://cloud.githubusercontent.com/assets/756501/5262550/f376ae88-79db-11e4-9f24-6e736e94d909.png)
![image](https://cloud.githubusercontent.com/assets/756501/5262573/1dc0e49c-79dc-11e4-93c4-cd44e8f25085.png)

I completely killed the drag functionality with my HTML and CSS rewrite although if this moves forward I intend to add it back. It was nice to not have to worry about it as I modified stuff and the current implementation didn't work on touch devices anyways.

I was looking at using one of these libraries for drag and drop since they work on touch devices:
http://jimliu.github.io/angular-ui-tree/index.html - Has great Angular bindings. I successfully used it to reorder the toolbar of new questions (although I haven't committed that yet)
http://rubaxa.github.io/Sortable/
http://touchpunch.furf.com/

What are your guys' initial thoughts? I probably should have discussed it first but I got a piece of it working and just kept going. I think it is more user friendly but I would like to hear feedback. Also, since I enabled responsiveness, some of the other pages and the header don't scale correctly in narrower viewports. That would take a bit to go through and fix but it shouldn't be too bad and should be done anyway if we want the site to be responsive.

This is NOT ready to merge, I just wanted to share it.

Also the delete page btn is a little big but I was trying to figure out what to do with it. Maybe I'll split it into a separate button group...

OzzieOrca commented:
@adammeyer Can you take a look at this again when you get a chance? I think I completed my list of things that I still needed to do with this PR. I'm not sure what you think of the btn-info color (used on Preview Form and Edit below). It is using the `@lightblue` LESS color that was at the top of `style.less`
![image](https://cloud.githubusercontent.com/assets/756501/6125165/24cbf862-b0c8-11e4-86d7-7edbcb64b399.png)

adammeyer commented:
Looking good.  2 things:
1) Is the "Cru Profile" button clickable?  If not, remove the border the so it doesn't look like a button.
2) I think the color needs to be reduced on this page.  Try making the "Edit" and "Copy" buttons the same as the "Move" white.  Same with the "Preview Form" button.  Try it in white.

OzzieOrca commented:
What does Cru Profile do exactly? I'm assuming it links the field(s) with some already existing info in a database. Is this explained to the admin when they are creating questions? Would this be commonly used or would most people leave it on default? Do admins need to quickly see whether the profile is enabled for each question? Is the action used enough to pull the button out of the edit section that expands? Currently it is just displaying whether it is enabled. I could make it clickable if that would be useful.

I made the rest of the buttons white. I feel like Edit should stand out a little bit more since it is used for accessing a lot of features. I kinda like the btn-warning color but it's not used anywhere else in the app currently.

adammeyer commented:
It's supposed to save your answers you use in those fields, and have them automatically be set for any future events you register for.  But I've never seen it actually work.  I'll ask Ryan if the backed is even supporting it.

I don't think it needs to be displayed up top for everyone question.  Maybe if the user clicks "Edit" for that question.  Like how it is now: 
![image](https://cloud.githubusercontent.com/assets/4603906/6249053/9ee2dcb6-b752-11e4-9a49-1f985f0b7cfb.png)
What do you think?

OzzieOrca commented:
Does it just populate all blocks of the same type in the future?
I can put it wherever. If it's not used much I can put it back in the edit section. I think I moved it up cuz it was in the same row as delete and copy but now that I think about it there wasn't much of a reason when I moved it.

adammeyer commented:
All blocks of the same type that have "Cru Profile" selected. But you can only have one Cru Profile per question per event.

OzzieOrca commented:
Ok. So does putting it back in the edit section sound good? I was thinking about having that green text and icon be the label for the checkbox

adammeyer commented:
Yes, sounds good.  I'm waiting to hear back from Ryan on it.  I don't think it even is functioning, so it might get removed.  I'll let you know when I hear back.

adammeyer commented:
I like it.  I'm working on pushing this to stage now.

adammeyer commented on app/views/eventDetails/eventInformation.html:
```diff
@@ -1,39 +1,38 @@
-        <option value="America/Los_Angeles">Pacific</option>
-        <option value="America/Anchorage">Alaska</option>
-        <option value="Pacific/Honolulu">Hawaii</option>
+      <option value="America/New_York">Eastern</option>
```
Is there a reason you removed the indents here?

OzzieOrca replied:
That probably happened in a merge... I'll fix it

adammeyer commented on app/views/modals/loginDialog.html:
```diff
@@ -4,9 +4,16 @@
+    </p>
+    <p>
+      <span class="text-muted">or</span>
+      <a class="logo ng-scope" href="#/" ng-click="close()">Return to the Welcome Page</a>
```
This should say "Go to Welcome Page" or something like that.  Most of the time the users will not becoming from there.

adammeyer commented on app/views/modals/paymentsModal.html:
```diff
@@ -1,324 +1,329 @@
+        <div class="col-sm-7 stacked-spacing-col-sm">
+          <label>Type</label>
+          <select ng-model="newTransaction.paymentType" class="form-control">
+            <option value="" disabled selected>Choose one...</option>
```
Nice!

adammeyer commented on app/views/modals/paymentsModal.html:
```diff
@@ -1,324 +1,329 @@
-              <input ng-model="newTransaction.check.checkNumber" type="text" class="form-control" />
-            </div>
+          <div class="col-sm-3">
+            <label>Verification Code</label>
```
I think "Security Code" would be more common language.

adammeyer commented on app/index.html:
```diff
@@ -185,8 +187,8 @@ <h1 ng-bind="globalPage.title"></h1>
-        <div class="col-xs-10">
+      <div class="row spacing-above-md spacing-below-md">
+        <div class="col-sm-10 stacked-spacing-col-sm">
           <a ng-href="#/{{registerMode || 'register'}}/{{globalPage.confId}}/page/" ng-if="globalPage.confId">
```
A hyperlink underline is added when hovering over the event name.  That should be removed.

adammeyer commented:
Looking great @OzzieOrca!  One idea I had: Could the event blocks on the dashboard be smaller?  Currently I have 7 events on mine, and it seems like a lot space. Maybe like the 2nd one?
![shorter](https://cloud.githubusercontent.com/assets/4603906/6374738/d3c9cc44-bce0-11e4-8360-2a81ffe9ea3c.png)

What do you think?

OzzieOrca commented:
I think I fixed all the issues above.

With the dashboard events, I reduced the padding. Here's what it looks like now
sm device:
![image](https://cloud.githubusercontent.com/assets/756501/6379381/b9db29da-bce6-11e4-90ed-6fe10bc0594b.png)
xs device:
![image](https://cloud.githubusercontent.com/assets/756501/6379403/dc71a38e-bce6-11e4-9c7e-c61b4c4d1639.png)

I could also switch it so the "3 Registered" button is in a row with the title:
sm device (reduces space for title):
![image](https://cloud.githubusercontent.com/assets/756501/6379451/3db55dd4-bce7-11e4-8688-22571270576a.png)
xs device (I don't like how this breaks it up. I suppose it could somewhere else on xs devices):
![image](https://cloud.githubusercontent.com/assets/756501/6379464/575639d4-bce7-11e4-8dff-a831e3775c97.png)

adammeyer commented:
I like it on the same row as the title.  Could you float it right so there isn't as much space between it and the title?

OzzieOrca commented:
That actually works a lot better. Idk why I didn't try that before.
![image](https://cloud.githubusercontent.com/assets/756501/6380423/66e0b49a-bcee-11e4-8916-d1ca99bc36f2.png)

adammeyer commented:
Looks great!

adammeyer commented on app/views/eventDetails/paymentOptions.html:
```diff
@@ -1,135 +1,150 @@
-        <label>Payment Gateway:</label>
-        <select class="form-control" ng-model="conference.paymentGatewayType" ng-options="gateway.id as gateway.name for gateway in paymentGateways"></select>
-      </div>
+    <div class="col-xs-6">
```
I think this looks better as xs-12, sm-6.

OzzieOrca commented:
Hey @adammeyer how do you think this looks with the new filters?

Collapsed:
![image](https://cloud.githubusercontent.com/assets/756501/6457443/bdaf51ec-c12a-11e4-90ae-19bcbedc8170.png)
Opened:
![image](https://cloud.githubusercontent.com/assets/756501/6457439/af241112-c12a-11e4-956f-84e2cf214ed3.png)
xs devices:
![image](https://cloud.githubusercontent.com/assets/756501/6457470/1707709e-c12b-11e4-939a-9690034585b8.png)

ryancarlson commented:
I like the concept a lot!

On Tue, Mar 3, 2015 at 1:26 AM, Scotty Waggoner <notifications@github.com>
wrote:

> Hey @adammeyer <https://github.com/adammeyer> how do you think this looks
> with the new filters?
>
> Collapsed:
> [image: image]
> <https://cloud.githubusercontent.com/assets/756501/6457443/bdaf51ec-c12a-11e4-90ae-19bcbedc8170.png>
> Opened:
> [image: image]
> <https://cloud.githubusercontent.com/assets/756501/6457439/af241112-c12a-11e4-956f-84e2cf214ed3.png>
> xs devices:
> [image: image]
> <https://cloud.githubusercontent.com/assets/756501/6457470/1707709e-c12b-11e4-939a-9690034585b8.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CruGlobal/conf-registration-web/pull/265#issuecomment-76893830>
> .
>



-- 
*Ryan Carlson*
Application Development Team
ryan.t.carlson@cru.org

adammeyer commented:
Looks great scotty!
